ios/ImagePickerManager.m

```diff
@@ -203,14 +203,27 @@ - (void)launchImagePicker:(RNImagePickerTarget)target
         }];
     }
     else { // RNImagePickerTargetLibrarySingleImage
-        [self checkPhotosPermissions:^(BOOL granted) {
-            if (!granted) {
-                self.callback(@[@{@"error": @"Photo library permissions not granted"}]);
-                return;
-            }
-
+        // Wrapper for permission check and showing picker
+        void (^showPickerViewControllerIfPermitted)() = ^() {
+            [self checkPhotosPermissions:^(BOOL granted) {
+                if (!granted) {
+                    self.callback(@[@{@"error": @"Photo library permissions not granted"}]);
+                    return;
+                }
+                
+                showPickerViewController();
+            }];
+        };
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0
```

Comment:
I included this macro check as I don't want to assume someone will only be using this on Xcode 9 and up.